js/src/modals/ExecuteContract/DetailsStep/detailsStep.js

```diff
@@ -141,6 +142,24 @@ export default class DetailsStep extends Component {
           );
           break;
 
+        case 'bool':
+          const boolitems = ['false', 'true'].map((bool) => {
+            return (
+              <MenuItem
+                key={ bool }
+                value={ bool }
+                label={ bool }>{ bool }</MenuItem>
+            );
+          });
+          inputbox = (
+            <Select
```

Comment:
* Checkbox is semantically not the best option.

* Toggle could work but does not allow for undefined value (the current dropdown solution either).

* Radio buttons are good because you can have 2 states (true / false), make the entry mandatory and define neither or the states as default so the user must make a choice and he can do so in a single click.

This is a detail, the current implementation is functional.